doc/api/assert.md

```diff
@@ -2539,6 +2539,112 @@ assert.throws(throwingFirst, /Second$/);
 // AssertionError [ERR_ASSERTION]
 ```
 
+## `assert.matchObject(actual, expected[, message])`
+
+<!-- YAML
+added: REPLACEME
+-->
+
+* `actual` {any}
+* `expected` {any}
+* `message` {string|Error}
+
+Tests equivalence between the actual and expected parameters by performing
+a deep comparison, ensuring that all properties and their values are equal,
+allowing type coercion where necessary.
+
+**Strict assertion mode**
+
+An alias of \[`assert.matchObjectStrict()`]\[].
+
+```mjs
+import assert from 'node:assert';
+
+assert.matchObject({ a: 1 }, { a: 1 });
+// OK
```

Comment:
Why isn't `assert.deepEqual(...)` and `assert.deepStrictEqual(...)` sufficient for these cases?

Reply:
It was a bit confusing, I've updated the docs to have better examples